src/communication/stream_transfer_receiver.hpp

```diff
@@ -58,6 +58,9 @@ namespace cubstream
       stream_position m_last_received_position;
       cubthread::daemon *m_receiver_daemon;
       char m_buffer[MTU];
+
+    protected:
+      cubstream::stream::write_func_t m_write_action_function;
```

Comment:
why is this in the protected group?

Reply:
I put it as protected in order to allow usage from potential derived classes, since it has function semantic; do you think we should forbid this  (not allow usage from derived) ?